robotoff/app/core.py

```diff
@@ -132,6 +132,34 @@ def get_insights(
     return query.iterator()
 
 
+def get_images(
+    with_predicted: Optional[int] = 1,
+    barcode: Optional[str] = None,
+    server_domain: Optional[str] = None,
+    offset: Optional[int] = None,
+    count: bool = False,
+    limit: Optional[int] = 25,
+) -> Iterable[ImageModel]:
+    if server_domain is None:
+        server_domain = settings.OFF_SERVER_DOMAIN
+
+    where_clauses = [ImageModel.server_domain == server_domain]
+
+    if barcode:
+        where_clauses.append(ImageModel.barcode == barcode)
+
+    query = ImageModel.select()
+
+    if where_clauses:
+        query = query.where(*where_clauses)
+
+    if with_predicted == 1:
+        # return all images
+        return query.iterator()
```

Comment:
with_predicted will be used to build the query so, no return here.

To build the query you should read https://docs.peewee-orm.com/en/latest/peewee/relationships.html

Hint: it's about outer join and using is_null() in where clause.
Do not hesitate to pdb in a test where you have created some Image and ImagePrediction, to try out the queries.